### PR TITLE
istioctl: fix `istioctl operator remove` fail when not specify reversion

### LINKED ISSUE #42159: `istioctl operator remove` fail when not specify version
### Bug Description

I used the latest histoctl compiled by the code to unload the operator, and found that the execution failed.
```shell
$ istioctl operator remove
Operator controller is not installed in istio-operator namespace (no Deployment detected).
Aborting, use --force to override.
```
The operator is exist:
```
$ kubectl -nistio-operator get po
NAME                              READY   STATUS    RESTARTS   AGE
istio-operator-5dc9778588-fx66z   1/1     Running   0          5m4s
```
Debug:
```shell
$ istioctl operator remove --vklog=6
....
2022-11-25T10:11:52.050561Z	info	klog	GET https://127.0.0.1:6443/version?timeout=5s 200 OK in 1 milliseconds
2022-11-25T10:11:52.055798Z	info	klog	GET https://127.0.0.1:6443/apis/apps/v1/namespaces/istio-operator/deployments/istio-operator-default 404 Not Found in 4 milliseconds
Operator controller is not installed in istio-operator namespace (no Deployment detected).
Aborting, use --force to override.
```
The required deployment name is `isito-operator-default`

### Version

```prose
istioctl version: Unpublished version, latest code of master branch
```

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by my-git9**
**Please provide a description of this PR:**
Fix: https://github.com/istio/istio/issues/42159

### CODE REVIEW

**Comment by my-git9** [comment]
The test result:
```shell
./istioctl operator init  --tag 1.15.1 --hub docker.io/istio -r default
Operator controller is already installed in istio-operator namespace.
Upgrading operator controller in namespace: istio-operator using image: docker.io/istio/operator:1.15.1
Operator controller will watch namespaces: istio-system
✔ Istio operator installed
✔ Installation complete
$
$
$ kubectl -nistio-system get po
NAME                             READY   STATUS             RESTARTS      AGE
istio-operator-7fb9cc79d-8h4qm   0/1     CrashLoopBackOff   3 (35s ago)   79s
$
$ 
$ ./istioctl operator remove
Istio operator revision default will be removed from cluster, Proceed? (y/N) y
Removing Istio operator...
  Removed Deployment:istio-operator:istio-operator.
  Removed Service:istio-operator:istio-operator.
  Removed ServiceAccount:istio-operator:istio-operator.
  Removed ClusterRole::istio-operator.
  Removed ClusterRoleBinding::istio-operator.
✔ Removal complete
$ kubectl get po -n istio-operator
No resources found in istio-operator namespace.
```

**Comment by zirain** [comment]
can you add a test/e2e for this?

**Comment by my-git9** [comment]
> can you add a test/e2e for this?

Sorry, I didn't find e2e test cases for installing, uninstalling (including `istioctl install` `istioctl uninstall`  `istioctl operator init` `istioctl operator remove`) from the istio repository.

**Comment by zirain** [comment]
can you try something like https://github.com/istio/istio/blob/master/tests/integration/operator/uninstall_test.go

**Comment by my-git9** [comment]
> can you try something like https://github.com/istio/istio/blob/master/tests/integration/operator/uninstall_test.go

Thansks, I wiil try it later

**Comment by my-git9** [comment]
PTAL @istio/wg-test-and-release-maintainers @istio/release-managers @istio/wg-environments-maintainers

**Comment by my-git9** [comment]
PTAL @zirain @ericvn @howardjohn @litong01

**Review by zirain — APPROVED** [review]
lgtm, thanks

File: releasenotes/notes/42160.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+apiVersion: release-notes/v2
+kind: bug-fix
+area: istioctl
+issue:
+  - 42160
+releaseNotes:
+  - |
+    **Fixed** `istioctl operator remove` failed when not specify revision.
```
Comment (ericvn) [inline releasenotes/notes/42160.yaml:8]:
```suggestion
    **Fixed** `istioctl operator remove` failed when a revision was not specified.
```

Comment (my-git9) [reply releasenotes/notes/42160.yaml]: Updated, thanks

**Review by ericvn — APPROVED** [review]

**Pull request merged**